### PR TITLE
Clean up common/Types.h

### PR DESCRIPTION
**Pull request by hainest**

### CODE REVIEW

File: dyninstAPI/src/dynProcess.C [hunk 1]
```diff
@@ -55,6 +55,27 @@
...
+	// must be used instead!
+	char *Address_str (Address addr)
+	{
+		static int i=0;
+		i=(i+1)%_numaddrstrs;
+		if (sizeof(Address) == sizeof(int))
+			snprintf(_addrstr[i],19,"0x%08X",(unsigned int)addr);
+		else
+			snprintf(_addrstr[i],19,"0x%016lX",(unsigned long)addr);
+		return (_addrstr[i]);
```
Comment (kupsch) [inline dyninstAPI/src/dynProcess.C:75]:
```
static_assert(sizeof(Address) <= sizeof(unsigned long), "Address has too many bits");
snprintf(_addrstr[i], 19, "0x%0*lX", int(2 * sizeof(Address)), (unsigned long)addr);
return _addrstr[i];
```

**Review by kupsch — APPROVED** [review]
A couple small comments that don't have to be acted on.

File: common/h/dyntypes.h [hunk 2]
```diff
@@ -106,7 +110,7 @@ namespace Dyninst
 #endif
 #endif
 
-   int ThrIDToTid(Dyninst::THR_ID id);
+   inline int ThrIDToTid(Dyninst::THR_ID id) { return id; }
```
[the remark this answers is not in the document]
Comment (hainest) [reply common/h/dyntypes.h:113]: Actually, it's never used. pickaxe also shows no past usage- only when it was introduced. I'll just delete it.

[the remark this answers is not in the document]
Comment (kupsch) [reply common/h/dyntypes.h:113]: sounds good.

File: dyninstAPI/src/dynProcess.C (hunk 1 above)
Comment (hainest) [reply dyninstAPI/src/dynProcess.C]: Address is typedef'ed to long on all platforms now, so I just removed the first snprintf.

**Pull request merged**